### PR TITLE
4710 save vaccination

### LINKED ISSUE #4710: Administer dose modal
From encounter view of vaccine card, clicking the row should open a modal like this:

Screenshot 2024-08-30 at 10 59 15 AM

Should be able to select one of the vaccine items configured for the related vaccine course.

Batch (`Lot`) opens another modal - separate issue

### PR DESCRIPTION
**Pull request by lache-melvin**
Fixes #4710

# 👩🏻‍💻 What does this PR do?

Implements backend for saving a vaccination

## 💌 Any notes for the reviewer?

# 🧪 Testing

- [ ] Run backend tests
- [ ] From patient encounter (if you update code to include a correct dose ID on the Vaccinations tab)
- [ ] You can save a `Not Given` dose, after providing a reason
- [ ] Note you won't be able to save a `Given` dose yet, as stock line selection not yet implemented

# 📃 Documentation

- [x] **Part of an epic**: documentation will be completed for the feature as a whole

### CODE REVIEW

File: server/repository/src/mock/encounter.rs [hunk 1]
```diff
@@ -0,0 +1,16 @@
+use crate::EncounterRow;
+
+use super::{mock_patient, mock_program_a};
+
+pub fn mock_encounter_a() -> EncounterRow {
```
Comment (lache-melvin) [inline server/repository/src/mock/encounter.rs:5]: Apologies for PR bloat, added a bit of mock data for testing 😅

File: server/service/src/vaccination/insert/mod.rs [hunk 2]
```diff
@@ -0,0 +1,331 @@
...
+    pub comment: Option<String>,
+    pub given: bool,
+    pub stock_line_id: Option<String>,
+    pub not_given_reason: Option<String>,
+}
+
+pub fn insert_vaccination(
+    ctx: &ServiceContext,
+    store_id: &str,
+    input: InsertVaccination,
```
Comment (CarlosNZ) [inline server/service/src/vaccination/insert/mod.rs:46]: Similar to my comment on last PR, do we want to add some logic here to ensure the "created_date" can't be before the minimum age for the course? (Would probably justify an additional Error value - `VaccintationTooEarly`?)

Can do in another PR -- maybe make an issue if you think worth it?

**Review by CarlosNZ — APPROVED** [review]
Great, thanks. Be good to get this in.

Similar comment about an additional check to ensure created_date is allowed.

File: server/service/src/vaccination/insert/mod.rs (hunk 2 above)
Comment (lache-melvin) [reply server/service/src/vaccination/insert/mod.rs:46]: Per other PR, product decision is to leave this open for clinician to deliver the dose before the exact age if need be 🙏
